GeomIO/SharedIO.F90

```diff
@@ -13,9 +14,128 @@ module mapl3g_SharedIO
    public get_mapl_geom
    public create_time_variable
    public bundle_to_metadata
+   public esmf_to_pfio_type
+   public create_local_start
+   public create_global_count
+   public create_global_start
+   public create_file_shape
 
    contains
 
+   function create_file_shape(grid, field_shape, rc) result(file_shape)
+      integer, allocatable :: file_shape(:)
+      type(ESMF_Grid), intent(in) :: grid
+      integer, intent(in) :: field_shape(:)
+      integer, intent(out), optional :: rc
+
+      integer :: status, sz, ungr, tile_count
+      call ESMF_GridGet(grid, tileCount=tile_count, _RC)
+      sz = size(field_shape)
+      ungr = sz - 2
+      if (tile_count == 6) then
+         allocate(file_shape(sz+1))
+         file_shape(1:3) = [field_shape(1), field_shape(2), 1]
+         file_shape(4:4+ungr-1) = [field_shape(2+ungr:sz)]
+      else if (tile_count == 1) then
+         file_shape = field_shape
+      else 
+         _FAIL("unsupported grid")
+      end if
+
+      _RETURN(_SUCCESS)
+   end function create_file_shape
+
+   function create_global_start(grid, field_shape, time_index, rc) result(global_start)
```

Comment:
There is enough duplication in logic in the procedures below that I'm wondering if it might be useful to instead do the following:
1. Introduce a new type that contains global bounds and local offset
2. Have one function that produces this type.

Reply:
Seems reasonable. And needs to be generalized eventually for other ESMF geom types. I'll combine into a single type.